src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -143,11 +145,36 @@ internal void SetInterfaceMapsBySilo(Dictionary<SiloAddress, GrainInterfaceMap>
             RebuildFullGrainInterfaceMap();
         }
 
-        internal IList<SiloAddress> GetSupportedSilos(int typeCode)
+        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode)
         {
             return supportedSilosByTypeCode[typeCode];
         }
 
+        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode, int ifaceId, IList<ushort> versions)
+        {
+            var result = new List<SiloAddress>();
+
```

Comment:
Can we cache the lists and invalidate the cache upon a cluster state change. This will allocate on every placement?